drivers/sx126x/sx126x_netdev.c

```diff
@@ -450,9 +450,9 @@ static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
         return sizeof(uint16_t);
 
     case NETOPT_SYNCWORD:
-        assert(len <= sizeof(uint8_t));
-        sx126x_set_lora_sync_word(dev, *((uint8_t *)val));
-        return sizeof(uint8_t);
+        assert(len <= sizeof(uint16_t));
```

Comment:
You may also want to change the corresponding netopt documentation:

https://github.com/RIOT-OS/RIOT/blob/c739516ac4b8f6772ae71a546da05e1d64a7ec5d/sys/include/net/netopt.h#L732-L735

Reply:
I have a doubt regarding this, the netopt would have two possible values, what would be simpler, upgrading the netop to a 16bit value and have the sx127x ignore the upper bits, or handle it as a buffer? Thoughts @jia200x @aabadie ?

Reply:
Hmm actually I think I can revert the whole sync word change, what the pkg code is doing is keeping nibble 0 and 2 the same (always `0x4`) and just changing the other two. So this code was actually correct. I'll revert and re-test when I get access to hardware again (this afternoon)